### PR TITLE
fix(update-build-links): Improve title parsing

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Improve title parsing, using separator instead of fixed length

### Motivation
Prevent bad replace if we have more than 10 RC (see page [comparison](https://datadoghq.atlassian.net/wiki/pages/diffpagesbyversion.action?pageId=2889876360&selectedPageVersions=112&selectedPageVersions=111))

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

File: tasks/release.py [hunk 1]
```diff
@@ -786,7 +786,7 @@ def update_build_links(_ctx, new_version, patch_version=False):
     content = confluence.get_page_by_id(page_id=BUILD_LINKS_PAGE_ID, expand="body.storage")
 
     title = content["title"]
-    current_version = title[-11:]
+    current_version = title.split()[-1].strip()
```
Comment (pducolin) [inline tasks/release.py:789]: **❓ question**
What is an example of `title`?

**Review by CelianR — APPROVED** [review]
**🥜 nitpick:** ‏Maybe we can add a regex to be sure ? Something like `[0-9]\.[0-9]+\.[0-9]+`

File: tasks/release.py (hunk 1 above)
Comment (chouetz) [reply tasks/release.py:789]: The page title is supposed to be `Build links - <version>`

**Pull request merged**